Sources/Brave/WebFilters/AdblockResourceDownloader.swift

```
@@ -70,37 +83,70 @@ public actor AdblockResourceDownloader: Sendable {
   }
   
   /// Load cached data for the given resource. Ensures this is done on the MainActor
-  private func loadCachedData(for resource: ResourceDownloader.Resource) async {
-    if let fileURL = ResourceDownloader.downloadedFileURL(for: resource) {
-      let date = try? ResourceDownloader.creationDate(for: resource)
-      await handle(downloadedFileURL: fileURL, for: resource, date: date)
+  private func loadCachedData(for resource: BraveS3Resource) async {
+    do {
+      if let downloadResult = try ResourceDownloaderStream.downloadResult(for: resource) {
+        await handle(downloadResult: downloadResult, for: resource)
+      } else {
+        switch resource {
+        case .genericContentBlockingBehaviors:
+          do {
+            // This is a special case which we load a bundled file if the downloaded file is not present
+            try await ContentBlockerManager.shared.compileBundledRuleList(for: .blockAds)
+          } catch {
+            assertionFailure("A bundled file should not fail to compile")
```

Cached data not loading should not crash the app even in debug mode

---

It's not cached. It's bundled. Not sure how a bundled file should fail to load unless it's missing and needs to fixed right away.

---

The function name is literally `loadCachedData`…

---

Yes but its in the else if there is no cache. So if you want I can call it `loadCachedOrBundledData`